### PR TITLE
Parse floating-point value from percentages

### LINKED ISSUE #1725: Better method of parsing %ages
### Please describe the problem (or idea)
https://github.com/publiclab/image-sequencer/blob/d33afe09da371c8aa69a3e84027b030f47b476bf/src/modules/Resize/Resize.js#L5

The resize module parses %ages eg `50%` by just removing the last character `50% -> 50` but this is an issue since one can type `50% ` with space or just `50`. We need to find a better way to parse percentages in all modules.

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site.

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by adzo261**
Fixes #1725

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm run test-all`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [ ] ask `@publiclab/is-reviewers` for help, in a comment below
* [ ] Insert-step functionality is working correct as expected.
> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software
Please make sure to get at least two reviews before asking for merging the PR as that would make the PR more reliable on our part
Thanks!

### CODE REVIEW

File: src/modules/Resize/Resize.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const imagejs = require('imagejs'),
   pixelSetter = require('../../util/pixelSetter'),
   ndarray = require('ndarray');
 module.exports = function Resize(pixels, options) {
-  const resize_value = parseInt(options.resize.slice(0, -1));
+  const resize_value = parseFloat(options.resize);
```
Comment (HarshKhandeparkar) [inline src/modules/Resize/Resize.js:5]: So this will take both with and without % right?

Comment (adzo261) [reply src/modules/Resize/Resize.js:5]: Yes

**Comment by HarshKhandeparkar** [comment]
Would you mind adding a test for this?

**Comment by adzo261** [comment]
Sure. Will do it.

File: src/modules/Resize/Resize.js (hunk 1 above)
Comment (HarshKhandeparkar) [reply src/modules/Resize/Resize.js:5]: Ok.. Ty.

File: test/core/modules/resize.js [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
+},
+options2 = {
+  resize: '60 %'
+};
+options3 = {
+  resize: '90'
+};
+testModule('resize', {resize: '129%'}, benchmark);
+
+require('../templates/options-test')('resize', [option1, option2, option3], [benchmark, benchmark, benchmark]);
```
Comment (HarshKhandeparkar) [inline test/core/modules/resize.js:15]: If you resize by different values, shouldn't the benchmark images be different too?

Comment (adzo261) [reply test/core/modules/resize.js:15]: Hi, can you help me generate images for the benchmark? I tried some base64 png's but the options' test is failing. However, there is no issue with the parsing of percentages. So I assume it is with the way I serve images.

Comment (HarshKhandeparkar) [reply test/core/modules/resize.js:15]: You can generate the images via CLI (read the readme/docs), or in the [browser](https://beta.sequencer.publiclab.org) and then convert them to a data URI using an online tool (google it, its very easy :))

File: test/core/modules/resize.js [hunk 3]
```diff
@@ -1,4 +1,21 @@
...
 
-testModule('resize', {resize: '129%'}, benchmark);
\ No newline at end of file
+  benchmark1 = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAsAAAALCAYAAACprHcmAAAAAklEQVR4AewaftIAAAAkSURBVJXBAQEAMAiAME7/zN4Ksr2Bz5EEEkgggQQSSCCBBBIs6poCE8Zr7KAAAAAASUVORK5CYII=',
+
+  benchmark2 = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAYAAACNMs+9AAAAAklEQVR4AewaftIAAAAiSURBVI3BAQEAAAiAIPP/5uqCMAtHIJFEEkkkkUQSSSTRAzwDAhGkYPRhAAAAAElFTkSuQmCC',
+
+  benchmark3 = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAYAAAAGCAYAAADgzO9IAAAAAklEQVR4AewaftIAAAAdSURBVHXBAQEAMAiAME7/zN4Csr2Bz0GCBAkSJCwpbQIJAvmJUgAAAABJRU5ErkJggg==',
+
+  options1 = {
```
Comment (HarshKhandeparkar) [inline test/core/modules/resize.js:10]: Can you write these as option-benchmark image pairs?

Comment (HarshKhandeparkar) [reply test/core/modules/resize.js:10]: Not necessary though, just for readability.

Comment (adzo261) [reply test/core/modules/resize.js:10]: Sorry I didn't get it. Can you clarify more?

Comment (HarshKhandeparkar) [reply test/core/modules/resize.js:10]: Oops. Sorry I didn't read your message.
I meant
```js
const benchmarks = [
{
  options: {},
  benchmark: ''
},
{
  options: {},
  benchmark: ''
},
...
]
```

Comment (HarshKhandeparkar) [reply test/core/modules/resize.js:10]: It is not necessary so let's leave it :)

**Review by HarshKhandeparkar — APPROVED** [review]

**Review by jywarren — APPROVED** [review]
Nice tests! Thank you!!!

**Comment by HarshKhandeparkar** [comment]
Pinned issue section? I guess this is specific for plots2, isn't it?

**Pull request merged**